[Experienced] Бан юзера за комменты - PostService

ViiRusZ opened the pull request.

ViiRusZx committed:
[Experienced] Бан юзера за комменты - PostService

ViiRusZx committed:
[Experienced] Бан юзера за комменты - PostService

ViiRusZx committed:
дописал тест

AlekseyB86 commented on src/main/java/faang/school/postservice/config/redis/RedisProperties.java:
```diff
@@ -0,0 +1,16 @@
+    private Integer port;
+    @Value("${spring.data.redis.user-ban-topic}")
+    private String usersBanTopic;
+}
```
не проще эти данные в классе конфиг получать? зачем отдельный для них создавать?

ViiRusZ replied:
нет. топиков может быть много, как и сервак может быть не одним.

zaqqaz226 replied:
Вообще в идеале, этот класс делать record (так как у нас пропертя всегда финальные и не будут изменены после старта приложения), накинуть аннотацию @ConfigurationProperties(prefix = "spring.data.redis") (вроде так пишется аннотация) и оно будет автоматом туда ходить и заполнять.

zaqqaz226 replied:
Но надо чтобы у нас названия переменной (с camelCase) совпадало с названием переменной в пропертях

AlekseyB86 commented on src/main/java/faang/school/postservice/model/Comment.java:
```diff
@@ -45,4 +45,7 @@ public class Comment {
     private LocalDateTime updatedAt;
+
+    @Column(name = "verified")
+    private boolean verified;
```
если примитив boolean то и св-ва колонки nullable=false, не?

AlekseyB86 replied:
Кстати у меня стоит Boolean, можешь сломать логику моего метода, сделай лучше Boolean

ViiRusZ replied:
Без указания NOT NULL, колонка типа boolean по умолчанию может содержать NULL, так как это стандартное поведение для всех типов данных в PostgreSQL, включая boolean. тут вопрос в другом, по null выборку делать из базы нельзя, но так и быть поменяю)

AlekseyB86 commented on src/main/java/faang/school/postservice/service/post/PostService.java:
```diff
@@ -34,6 +44,7 @@
 @RequiredArgsConstructor
 @Validated
 public class PostService {
+    private static final Logger log = LoggerFactory.getLogger(PostService.class);
```
зачем Logger? ведь везде у нас @Slf4j

ViiRusZ replied:
точно, тут я не смотрел, а там аннотация не  накинута была.

AlekseyB86 commented on src/main/java/faang/school/postservice/service/post/PostService.java:
```diff
@@ -132,27 +143,32 @@ public boolean existsPost(Long postId) {
-        return postRepository.findByNotPublishedAndNotDeletedAndAuthorIdOrderCreatedAtDesc(userId).stream()
-                .map(postMapper::toDraftDtoFromPost)
-                .toList();
+        return postRepository.findByNotPublishedAndNotDeletedAndAuthorIdOrderCreatedAtDesc(userId).stream().map(postMapper::toDraftDtoFromPost).toList();
```
смайл рука лицо, смысл?))

ViiRusZ replied:
хард варп... а так то твой код ахахаха...

AlekseyB86 commented on src/main/java/faang/school/postservice/sheduler/CommenterBanner.java:
```diff
@@ -0,0 +1,20 @@
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
```
там Logger, здесь @Slf4j)))

ViiRusZ replied:
если вглядеться можно увидеть :)

AlekseyB86 reviewed (approved):
Супер

ViiRusZx committed:
поправил немного

ViiRusZx committed:
Merge branch 'refs/heads/Cyclop-master-stream7' into feauture-BJS2-41234-PostService

# Conflicts:
#	src/main/java/faang/school/postservice/PostServiceApp.java
#	src/main/java/faang/school/postservice/model/Comment.java
#	src/main/java/faang/school/postservice/repository/CommentRepository.java
#	src/main/java/faang/school/postservice/service/post/PostService.java

ViiRusZx committed:
посмотрел как работает автосоздание Pojo кслассов через схемы, немного поправил логику, переписал тесты

22maksim commented on src/main/java/faang/school/postservice/sheduler/CommenterBanner.java:
```diff
@@ -12,7 +12,7 @@
     private final PostService postService;
 
-    @Scheduled(cron = "${CommentBanner.cron.expression.ban-comments}")
+    @Scheduled(fixedDelay = 5000)
```
может вернуть?

ViiRusZx committed:
посмотрел как работает автосоздание Pojo кслассов через схемы, немного поправил логику, переписал тесты

ViiRusZx committed:
Merge branch 'refs/heads/Cyclop-master-stream7' into feauture-BJS2-41234-PostService

# Conflicts:
#	src/main/java/faang/school/postservice/service/post/PostService.java
#	src/test/java/faang/school/postservice/service/post/PostServiceTest.java

ViiRusZx committed:
что то доделал

ViiRusZx committed:
что то доделал

zaqqaz226 commented:
Оставил тебе там коммент как можно сделать конфиги еще красивее и лучше. Изменения на твое усмотрение.

zaqqaz226 commented:
А да, может ругаться на конфиги, если решишь поменять, тогда надо будет в классе сервиса накинуть @EnableConfigurationProperties(YourClassWithProps.class)

zaqqaz226 commented:
Ну и какие то конфликты на ветке

ViiRusZx committed:
сделал рекорд

ViiRusZx committed:
Merge branch 'refs/heads/Cyclop-master-stream7' into feauture-BJS2-41234-PostService

# Conflicts:
#	src/main/java/faang/school/postservice/PostServiceApp.java
#	src/main/java/faang/school/postservice/service/post/PostService.java
#	src/test/java/faang/school/postservice/service/post/PostServiceTest.java

ViiRusZx committed:
сделал рекорд, решил конфликты

zaqqaz226 commented:
Появились конфликты, а так можно катить

22maksim committed:
Merge branch 'Cyclop-master-stream7' into feauture-BJS2-41234-PostService

22maksim committed:
Update PostService.java

22maksim committed:
Update PostServiceTest.java

22maksim committed:
Update PostServiceTest.java

22maksim committed:
Update PostService.java

22maksim committed:
Update PostService.java

22maksim committed:
Update PostService.java

22maksim committed:
Update PostService.java

22maksim committed:
Update PostService.java

22maksim committed:
Update PostService.java

22maksim committed:
Update RedisConfig.java

22maksim committed:
Update RedisConfig.java

НУ СЕЙЧАС ТО ВСЁ??? ААААААААААААААААААААААААААААААААААААААААААААААААААААААААААААААААААААААААААААААААА

22maksim committed:
Update PostServiceTest.java

Надеюсь всё ;-)

22maksim committed:
Update PostServiceTest.java

22maksim committed:
Update PostService.java

22maksim committed:
Update MessageSenderForUserBanImpl.java

22maksim committed:
Update RedisConfig.java

22maksim committed:
Update application.yaml

22maksim committed:
Update application.yaml

22maksim committed:
Update application.yaml

22maksim committed:
Update application.yaml
